Mathlib/Data/Finset/Prod.lean

```diff
@@ -0,0 +1,448 @@
+/-
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johannes Hölzl, Mario Carneiro, Oliver Nash
+
+! This file was ported from Lean 3 source module data.finset.prod
+! leanprover-community/mathlib commit 9003f28797c0664a49e4179487267c494477d853
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Finset.Card
+
+/-!
+# Finsets in product types
+
+This file defines finset constructions on the product type `α × β`. Beware not to confuse with the
+`Finset.prod` operation which computes the multiplicative product.
+
+## Main declarations
+
+* `Finset.product`: Turns `s : Finset α`, `t : Finset β` into their product in `Finset (α × β)`.
+* `Finset.diag`: For `s : Finset α`, `s.diag` is the `Finset (α × α)` of pairs `(a, a)` with
+  `a ∈ s`.
+* `Finset.offDiag`: For `s : Finset α`, `s.offDiag` is the `Finset (α × α)` of pairs `(a, b)` with
+  `a, b ∈ s` and `a ≠ b`.
+-/
+
+
+open Multiset
+
+variable {α β γ : Type _}
+
+namespace Finset
+
+/-! ### prod -/
+
+
+section Prod
+
+variable {s s' : Finset α} {t t' : Finset β} {a : α} {b : β}
+
+/-- `product s t` is the set of pairs `(a, b)` such that `a ∈ s` and `b ∈ t`. -/
+protected def product (s : Finset α) (t : Finset β) : Finset (α × β) :=
+  ⟨_, s.nodup.product t.nodup⟩
+#align finset.product Finset.product
+
+--Porting note: Change notation from  "×ᶠ" to "×ᶠ" to avoid ambiguity
```

Comment:
Is there a difference between those that I'm not seeing?

Reply:
Overzealous search and replace is responsible for this